### PR TITLE
feat: add some fork choice values to BeaconChain.

### PR DESCRIPTION
**Pull request by mpaulucci**
We don't want to have to calculate the new head everytime it is requested since it's expensive. Let's store a "cache" (better names welcomed) that gets updated every time we get a new block.

### CODE REVIEW

File: lib/lambda_ethereum_consensus/beacon/beacon_chain.ex
Comment (mpaulucci) [inline lib/lambda_ethereum_consensus/beacon/beacon_chain.ex:25]: could prepend all these values with `last`, i.e `last_finalized_root`

**Review by MegaRedHand — COMMENTED** [review]
LGTM. I'll let @Arkenan have a final look.

File: lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex
Comment (mpaulucci) [inline lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex:75]: I've had to turn off db because otherwise the `get_head` function fails.

**Review by MegaRedHand — APPROVED** [review]
LGTM

File: lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex
Comment (MegaRedHand) [reply lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex:75]: I'll take a look

File: lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex
Comment (Arkenan) [inline lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex:166]: Instead of calling a `cast` to update the genserver, we can match the message against `{_ref, new_fork_choice_cache}` and receive the task result directly.

File: lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex
Comment (Arkenan) [inline lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex:226]: We can return a map or a tuple with this and it can be caught by handle_info instead of calling a cast.

**Review by Arkenan — APPROVED** [review]
LGTM! I suggested a minor change, but otherwise, feel free to merge.

**Review by Arkenan — APPROVED** [review]
LGTM after re-review and offline discussion.

**Pull request merged**
